lib/sles4sap_publiccloud.pm

```diff
@@ -182,6 +184,70 @@ sub sles4sap_cleanup {
 
 }
 
+=head2 get_job_state
+    Get whether a specified job is still running or not
+=cut
+
+sub get_job_state {
+    my ($job_id) = @_;
+
+    my $url = "https://openqa.suse.de/api/v1/jobs/$job_id";
+
+    my $json_data = script_output("curl -s '$url'");
+
+    my $job_data = eval { decode_json($json_data) };
+    if ($@) {
+        warn "Failed to decode JSON data for job $job_id: $@";
+        return 0;    # Assume job is still running if we can't get its state
+    }
+
+    my $job_state = $job_data->{job}->{state} // '';
+
+    if ($job_state eq 'running') {
+        return 0;    # Job is still running
+    }
+    else {
+        return 1;    # Job has stopped
+    }
+}
+
+
+=head2 get_peerings_azure
+    Get active peerings for Azure jobs
+=cut
+
+sub get_peerings_azure {
+    my ($self) = @_;
+    my $cmd = 'az network vnet peering list -g dsdm0we --vnet-name vnet-dsdm0we --query "[].name"';
```

Comment:
We do not have IBSm in any code, we are using `IBSM_RG` to inject it. Please also take care that we do not have `IBSM_RG` in qesapdeployment.pm but only in `sles4sap_publiccloud.pm` layer calling `qesap` low lib layer. Please tr to maintain this separation.

Reply:
Done